resolver/resolver.go

```diff
@@ -29,9 +29,7 @@ type Resolver struct {
 	rs      *records.RecordGenerator
 	rsLock  sync.RWMutex
 	rng     *rand.Rand
-
-	// pluggable external DNS resolution, mainly for unit testing
-	extResolver exchanger.Exchanger
+	exchanger.Forwarder
```

Comment:
I'm pro making this private, but you have to convince me of the **confusion** associated with the `Forward` method. :-)